### ISSUE TITLE
beamercolorbox: unexpected behavior

### ISSUE THREAD

**Issue by josephwright** [josephwright/beamer#1]
Hi,

not really sure if this is a bug or not, but the behavior is somewhat unexpected to me: When invoking
    
    \begin{beamercolorbox}[leftskip=1cm,right]{}
    [...]
    \end{beamercolorbox}
    
the leftskip is entirely ignored. When doing
    
    \begin{beamercolorbox}[right,leftskip=1cm]{}
    [...]
    \end{beamercolorbox}
    
the text `[...]` is **not** set flush right.

Cheers,
Nico

----------------------------------------
- Bitbucket: https://bitbucket.org/rivanvx/beamer/issue/1
- Originally reported by: Nico Schlömer
- Originally created at: 2010-04-26T22:37:27.173

**Comment by josephwright** [comment]
_From Vedran Miletić on 2010-04-26_

Seems like a bug. I will look into it.

**Comment by josephwright** [comment]
_From Vedran Miletić on 2010-04-27_



Those options seem to be in conflict, because:
````
\define@key{beamercolbox}{leftskip}{\def\beamer@colbox@ls{#1}}
\define@key{beamercolbox}{right}[]{\def\beamer@colbox@ls{0pt plus1fill}\def\beamer@colbox@rs{0pt}}
````
I have yet to think of an intelligent way to fix this.

**Comment by josephwright** [comment]
_From Vedran Miletić on 2010-05-02_

I had to backout the changeset, it breaks a lot of other things.

**Comment by josephwright** [comment]
_From Vedran Miletić on 2010-05-02_

 had to backout the changeset, it breaks a lot of other things.

**Comment by josephwright** [comment]
_From Vedran Miletić on 2010-05-04_

Hm, that's too bad.

Does that mean that the first argument is actually meant to be ignored, and that many things relied on this? Would it be desirable to fix this then, after all?

For me it wouldn't really make a big difference if there was another way to get left/right skips and horizontal text alignment.

Anyway, also the fix that forces you to put the skips first, and then the justification could potentially confuse users who don't know about this rule and happen to declare the arguments in the wrong order.

In any case: nasty.

--Nico

**Comment by josephwright** [comment]
From Vedran Miletić on 2010-05-31

We should really fix it for next release. I will think of something over the weekend.

**Comment by josephwright** [comment]
From Vedran Miletić on 2010-06-06

Well, let's try this way: I have put in the code that supports this:

    1. `left=2cm` will put text flush left and adjust right skip to 2cm
    2. `right=2cm` will put text flush right and adjust left skip to 2cm
    3. `center=2cm` will center text and adjust both skips to 2cm 

I hope no-one will ask for an asymetric margins in center :-)

For now, this change will not be documented in user guide since I hope a better interface will eventually be come out, when I learn a bit more (La)TeX programming. But feel free to use it.

**Comment by josephwright** [comment]
_From Nico Schlömer  on 2010-06-06_

Does *not* work for me. The code
````
\begin{beamercolorbox}[right=\leftskip]{frame title}%
\insertframetitle\par%
\end{beamercolorbox}%
````
employs the skip, but sets the title flush left.

**Comment by josephwright** [comment]
_From Vedran Miletić on 2010-06-13_

What do you expect it to do?

**Comment by josephwright** [comment]
_From Nico Schlömer on 2010-06-13_

Err -- I thought your previous comment suggests that `right=something` will make sure the text is actually set flush right.

**Comment by josephwright** [comment]
_From Vedran Miletić on 2010-06-13_

Actually, at least here, it does when you use a constant number (e.g. 2cm).

**Comment by josephwright** [comment]
_From Nico Schlömer on 2010-06-15_

Same for me, but I just noticed that when putting a variable such as `\leftskip` as in
````
\leftskip=\ulogowidth%
\advance\leftskip by\ulogopadding%
````
the theme will plot a literal "plus1fill". Bug?

**Comment by josephwright** [comment]
_From Vedran Miletić on 2010-06-17_

Likely. However, I'm not Till and I don't have his TeXperience that would be needed to fix it. I tried messing with `\expandafter` with no change. Perhaps Joseph can assist?

**Comment by josephwright** [comment]
_From Vedran Miletić on 2010-10-07_

I will fix this when I learn how to do it. For now, it will stay as it is, no code I implemented already is going to be removed.

**Comment by josephwright** [comment]
_From Joseph Wright on 2012-02-23_

Fixed by e89382c
